core/src/main/java/org/eclipse/hono/util/AdapterInstanceStatus.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.util;
+
+/**
+ * The adapter instance status.
+ */
+public enum AdapterInstanceStatus {
+    /**
+     * Adapter instance is alive.
+     */
+    ALIVE,
+    /**
+     * Adapter instance has probably been terminated.
+     */
+    SUSPECTED,
```

Comment:
The *KubernetesBasedAdapterInstanceStatusService#getStatus()* method contains the following comment for the case when it returns SUSPECTED:
> (if the) container is new and the watcher hasn't informed us about it yet (watcher could be currently reconnecting internally)

Base don that, something like PENDING would better fit the purpose here ...

Reply:
PENDING would have this connotation of "container just being started", but that is just the unlikely scenario.
I've removed the above comment and put a new one in the SUSPECTED block:
```
                // container hasn't explicitly been marked as terminated
                // so it could either have been terminated before this service got started
                // or, in an unlikely case, could be newly started and alive, and the watcher hasn't informed us about it yet
                // (watcher could be currently reconnecting internally)
```

Reply:
Ok, can we then use something like INDETERMINATE? SUSPECTED sounds like the pod had committed a crime ... ;-)

Reply:
I've used SUSPECTED thinking about the same term being used for JGroups/Infinispan cluster nodes that are believed to be dead. INDETERMINATE would be quite similar to UNKNOWN.

Reply:
Then maybe SUSPECTED_DEAD?

Reply:
I'm fine with that. I've renamed it accordlingly, also renaming TERMINATED to DEAD as a consequence.